File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_till_here = 0
        length = len(nums)
        for i in range(length):
            if(max_till_here < i):
                return False
            max_till_here = max(max_till_here, i + nums[i])
        if max_till_here >= length - 1:
            return True
        return False

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxReach = 0;
        int n = nums.length;
        
        for (int i = 0; i < n; i++) {
            if (maxReach < i) {
                return false;
            } else if(maxReach >= n) {
                return true;
            }
            
            maxReach = Math.max(i + nums[i], maxReach);
        }
        return true;
    }
}